cumulus/parachains/integration-tests/emulated/tests/assets/asset-hub-westend/src/tests/aliases.rs

```diff
@@ -0,0 +1,181 @@
+	let fees = WESTEND_ED * 10;
+
+	PenpalA::mint_foreign_asset(
+		<PenpalA as Chain>::RuntimeOrigin::signed(PenpalAssetOwner::get()),
+		Location::parent(),
+		origin.clone(),
+		fees * 10,
+	);
+
+	// Aliasing different account on different chains
+	test_cross_chain_alias!(
+		vec![
+			// between BH and AH: denied
+			(BridgeHubWestend, AssetHubWestend, TELEPORT_FEES, DENIED),
+			// between Collectives and AH: denied
+			(CollectivesWestend, AssetHubWestend, TELEPORT_FEES, DENIED),
+			// between Coretime and AH: denied
+			(CoretimeWestend, AssetHubWestend, TELEPORT_FEES, DENIED),
+			// between People and AH: denied
+			(PeopleWestend, AssetHubWestend, TELEPORT_FEES, DENIED),
+			// between Penpal and AH: denied
+			(PenpalA, AssetHubWestend, RESERVE_TRANSFER_FEES, DENIED)
+		],
+		origin,
+		target,
+		fees
+	);
+}
+
+#[test]
+fn aliasing_child_locations() {
+	use AssetHubWestendXcmConfig as XcmConfig;
+	// Allows aliasing descendant of origin.
+	let origin = Location::new(1, X1([PalletInstance(8)].into()));
+	let target = Location::new(1, X2([PalletInstance(8), GeneralIndex(9)].into()));
+	assert!(<XcmConfig as xcm_executor::Config>::Aliasers::contains(&origin, &target));
+	let origin = Location::new(1, X1([Parachain(8)].into()));
+	let target =
+		Location::new(1, X2([Parachain(8), AccountId32 { network: None, id: [1u8; 32] }].into()));
+	assert!(<XcmConfig as xcm_executor::Config>::Aliasers::contains(&origin, &target));
+	let origin = Location::new(1, X1([Parachain(8)].into()));
+	let target = Location::new(1, X3([Parachain(8), PalletInstance(8), GeneralIndex(9)].into()));
+	assert!(<XcmConfig as xcm_executor::Config>::Aliasers::contains(&origin, &target));
+
+	// Does not allow if not descendant.
+	let origin = Location::new(1, X1([PalletInstance(8)].into()));
+	let target = Location::new(0, X2([PalletInstance(8), GeneralIndex(9)].into()));
+	assert!(!<XcmConfig as xcm_executor::Config>::Aliasers::contains(&origin, &target));
+	let origin = Location::new(1, X1([Parachain(8)].into()));
+	let target =
+		Location::new(0, X2([Parachain(8), AccountId32 { network: None, id: [1u8; 32] }].into()));
+	assert!(!<XcmConfig as xcm_executor::Config>::Aliasers::contains(&origin, &target));
+	let origin = Location::new(1, X1([Parachain(8)].into()));
+	let target = Location::new(0, X1([AccountId32 { network: None, id: [1u8; 32] }].into()));
+	assert!(!<XcmConfig as xcm_executor::Config>::Aliasers::contains(&origin, &target));
+	let origin = Location::new(1, X1([AccountId32 { network: None, id: [1u8; 32] }].into()));
+	let target = Location::new(0, X1([AccountId32 { network: None, id: [1u8; 32] }].into()));
+	assert!(!<XcmConfig as xcm_executor::Config>::Aliasers::contains(&origin, &target));
+}
+
+#[test]
+fn asset_hub_root_aliases_anything() {
+	use AssetHubWestendXcmConfig as XcmConfig;
+
+	// Does not allow local/AH root to alias other locations.
+	let origin = Location::new(1, X1([Parachain(1000)].into()));
```

Comment:
The asset hub location should be `(0, [])` from the perspective of asset hub, no?

Reply:
Fixed in follow-up PR in commit: https://github.com/paritytech/polkadot-sdk/pull/6336/commits/5491a125eabd215807a80f02828345045ff3816b

Please help with review of this follow-up PR too: https://github.com/paritytech/polkadot-sdk/pull/6336